orchestration/src/main/java/com/sap/ai/sdk/orchestration/spring/OrchestrationChatModel.java

```diff
@@ -29,28 +38,52 @@
  */
 @Beta
 @Slf4j
-@RequiredArgsConstructor
-public class OrchestrationChatModel implements ChatModel {
-  @Nonnull private OrchestrationClient client;
+public class OrchestrationChatModel extends DefaultToolCallingManager implements ChatModel {
```

Comment:
(Question/Major)

Can you explain why you went with inheritance, instead of using composition?

Reply:
If you were using composition, then you'd instantiate `DefaultToolCallingManager` using the builder pattern.
As a result you wouldn't need the additional dependency
```java
DefaultToolCallingManager.builder(). [...]
```